native/packages/react-native-bpk-component-touchable-overlay/src/BpkTouchableOverlay.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { View, ViewPropTypes, TouchableWithoutFeedback, Platform, StyleSheet } from 'react-native';
+
+const tokens = Platform.select({
+  ios: () => require('bpk-tokens/tokens/ios/base.react.native.common.js'), // eslint-disable-line global-require
+  android: () => require('bpk-tokens/tokens/android/base.react.native.common.js'), // eslint-disable-line global-require
+})();
+
+const styles = StyleSheet.create({
+  overlay: {
+    flex: 1,
+    position: 'absolute',
+    top: 0,
+    right: 0,
+    bottom: 0,
+    left: 0,
+    backgroundColor: tokens.colorGray900,
+    opacity: 0,
+  },
+  showOverlay: {
+    opacity: 0.15,
+  },
+});
+
+const BpkTouchableOverlay = (props) => {
+  const {
+    children,
+    style,
+    onPressIn,
+    onPressOut,
+    ...rest
+  } = props;
+
+  let overlayRef = null;
+
+  return (
+    <TouchableWithoutFeedback
+      {...rest}
+      onPressIn={() => {
+        overlayRef.setNativeProps({ style: [styles.overlay, styles.showOverlay] });
+        if (onPressIn) { onPressIn(); }
+      }}
+      onPressOut={() => {
+        overlayRef.setNativeProps({ style: [styles.overlay] });
+        if (onPressOut) { onPressOut(); }
+      }}
+    >
+      <View style={style}>
+        {children}
+        <View
+          style={styles.overlay}
+          ref={(ref) => { overlayRef = ref; }}
+        />
+      </View>
+    </TouchableWithoutFeedback>
+  );
+};
+
+BpkTouchableOverlay.propTypes = {
+  children: PropTypes.node.isRequired,
+  style: ViewPropTypes.style,
```

Comment:
Should we have an explicit `overlayColor` prop like `TouchableHighlight`?

Reply:
Do we also want to expose the ability to control the overlay opacity?

Reply:
I've added the tokens. I don't think we wanna provide the ability to customise the overlay opacity... @jamesf3rguson ?